### PR TITLE
Update Redox

### PR DESCRIPTION
**Pull request by Snowiiii**
- [X] Tested on all platforms changed
- [X] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [X] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [X] Created or updated an example program if it would help users understand this functionality
- [X] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

`orbclient` has dropped `redox_syscall` dependency

Should close https://github.com/rust-windowing/winit/pull/3181

### CODE REVIEW

**Review by kchibisov — COMMENTED** [review]
You should update the `deny.toml` as well.

**Comment by Snowiiii** [comment]
done

**Review by jackpot51 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,8 @@ Unreleased` header.
 - **Breaking:** Rename `VideoMode` to `VideoModeHandle` to represent that it doesn't hold static data.
 - **Breaking:** No longer export `platform::x11::XNotSupported`.
 - **Breaking:** Renamed `platform::x11::XWindowType` to `platform::x11::WindowType`.
+- On Redox, Update `redox_syscall` from 0.3 to 0.4.1
+- On Redox, Update `orbclient` from 0.3.42 to 0.3.47
```
Comment (daxpedda) [inline CHANGELOG.md:32]: If this isn't exposed to the user, it doesn't need a changelog entry.

Comment (Snowiiii) [reply CHANGELOG.md:32]: Should i remove it?

**Pull request merged**